src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -837,6 +837,34 @@ public override ForEachStatementInfo GetForEachStatementInfo(CommonForEachStatem
                 enumeratorInfoOpt.CurrentConversion);
         }
 
+        public override DeconstructionInfo GetDeconstructionInfo(AssignmentExpressionSyntax node)
+        {
+            var boundDeconstruction = (BoundDeconstructionAssignmentOperator)GetUpperBoundNode(node);
```

Comment:
Does this throw a cast exception if the caller passes an assignment that is not a deconstruction assignment? Is that the desired behavior?

Reply:
Good question. It's the same behavior as `GetForEachStatementInfo`, but does seem wrong.
For comparison, `GetAwaitExpressionInfo` returns `default` in such cases. I'll change foreach and deconstruction methods to do the same.


---
In reply to: [149514253](https://github.com/dotnet/roslyn/pull/22934#discussion_r149514253) [](ancestors = 149514253)

Reply:
Hum, actually `GetForEachStatementInfo` is fine. The problem is just with assignment (which may or may not be a deconstruction).

---
In reply to: [149516672](https://github.com/dotnet/roslyn/pull/22934#discussion_r149516672) [](ancestors = 149516672,149514253)